provision/acc_provision/templates/aci-containers.yaml

```diff
@@ -1542,9 +1542,11 @@ data:
             ,"statistics" : { "mode" : "off" }
           {% endif %}
         },
+        {% if config.kube_config.opflex_agent_prometheus %}
```

Comment:
It seems like this check is evaluating to False when the value of opflex_agent_prometheus is False. Hence we should not be doing this check in this place but rather should be just setting the value below (either its True or False). By doing the check here we are populating the key:value below only in the True case.

Reply:
@snaiksat got confused by the testdatafiles.. but your logic makes sense as I cross-checked it with below sample of code as well.. 

```
d = {"A":1, "B":0, "C":None}
print(d)
print(type(d))
for i,j in d.items():
    print i,j
if d["A"]:
    print d["A"]
if d["B"]:
    print d["B"]
if d["C"]:
    print d["C"]
```

I was with the assumption that if checks whether key is present or not more than checking the value but thats not the case as you can see for the above snippet of code. It checks the values is None/0 or not.

I will remove the if check here, then it should be fine I guess.. as it would set itself to true or false depending on case.